### PR TITLE
ENG-6589 - Refactor to Async/Await Pattern

### PR DESCRIPTION
**Pull request by kevinsites**

### CODE REVIEW

**Review by priya-xavier — APPROVED** [review]

File: NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+    }
+
+//    getJsonPayLoad - not sure how to mock
+//    resetJsonPayLoad - not sure how to mock
+
+    //    captureEvent - not sure how to mock Application context
+    @Test
+    @Ignore("Ignore until mocked Application")
+    fun testCaptureEvent() {
+//        = runTest
```
Comment (kishisaka-nid) [inline NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt:395]: remove the unused method

File: NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+//        events = getDataStoreInstance().getAllEvents()
+//        println("EVENTA: ${events.toString()} - ${events.count()} - ${events}")
+//        advanceUntilIdle()
+//        assertEquals(1, events.count())
+    }
+
+    //    formSubmit - Need to mock Datastore
+    @Test
+    @Ignore("Ignore until mocked Datastore")
+    fun testFormSubmit() {
```
Comment (kishisaka-nid) [inline NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt:414]: remove unused method

File: NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+    private fun setNeuroIDInstance() {
+        val neuroId = NeuroID.Builder(
+            null,
+            "key_test_fake1234"
+        ).build()
+        NeuroID.setNeuroIdInstance(neuroId)
+    }
+
+    private fun setNeuroIDMockedLogger(
+        errorMessage: String = "",
```
Comment (kishisaka-nid) [inline NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt:55]: This is cool!

**Review by kishisaka-nid — COMMENTED** [review]
nice! 🚢 it!

**Review by kishisaka-nid — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Comment by kishisaka-nid** [comment]
Need to add the dependency 
`testImplementation 'org.jetbrains.kotlinx:kotlinx-coroutines-test:1.7.3'`
in the app build.gradle

**Review by kishisaka-nid — DISMISSED** [review]
LGTM. Nice! 🚢 it!

File: NeuroID/src/test/java/com/neuroid/tracker/NIDDataStoreManagerUnitTests.kt [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.neuroid.tracker.events.INPUT
+import com.neuroid.tracker.models.NIDEventModel
+import com.neuroid.tracker.storage.NIDDataStoreManagerImp
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.runs
```
Comment (kishisaka-nid) [inline NeuroID/src/test/java/com/neuroid/tracker/NIDDataStoreManagerUnitTests.kt:11]: nit, unused import here

**Review by priya-xavier — APPROVED** [review]

**Review by cometman — DISMISSED** [review]

**Review by kishisaka-nid — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Review by cometman — APPROVED** [review]

**Pull request merged**